src/RedemptionPool.sol

```diff
@@ -98,74 +140,98 @@ contract RedemptionPool is Ownable {
+
+        // Redeem the user's cUSDC tokens for USDC tokens
+        // and transfer the USDC tokens to the user's address
+        uint256 redeemResult = ICERC20(CUSDC).redeem(_amount);
+        if (redeemResult != 0)
+            revert RedemptionErrors.USDCRedeemFailed(redeemResult);
+
+        uint256 usdcAmount = (ICERC20(CUSDC).exchangeRateStored() * _amount) /
+            1e20;
+
+        IERC20(USDC).safeTransfer(msg.sender, usdcAmount);
+
+        // Adjust the user's and the cumulative tally of claimed cUSDC
+        _userClaims[msg.sender] += _amount;
+        totalCUSDCWithdrawn += _amount;
+        emit Claim(msg.sender, _amount);
     }
 
     /////////////////////////////////////////////////////////////////////////////
     //                              Permissioned funcs                         //
     /////////////////////////////////////////////////////////////////////////////
 
-    /// @notice Pulls assets from the msig
-    /// @param amount amount of cUSDC to pull
-    function depositCUSDC(uint256 amount) external onlyOwner {
-        require(amount > 0, "amount must be greater than 0");
-        require(IERC20(CUSDC).transferFrom(msg.sender, address(this), amount), "transfer failed");
-        totalAssetsDeposited += amount;
-        emit CUSDCDeposit(totalAssetsDeposited);
+    /// @notice Pulls assets from the DAO msig
+    /// @param _amount amount of cUSDC to pull
+    function depositCUSDC(uint256 _amount) external onlyOwner {
+        // Transfer cUSDC from the caller to this contract
+        IERC20(CUSDC).safeTransferFrom(msg.sender, address(this), _amount);
+
+        totalCUSDCDeposited += _amount;
+        emit CUSDCDeposit(totalCUSDCDeposited);
     }
 
-    /// @notice Allow to withdraw any tokens except GRO back to the owner
+    /// @notice Allow to withdraw any tokens except GRO back to the owner, as long as the deadline has not passed
     /// @param _token address of the token to sweep
-    function sweep(address _token) external onlyOwner {
+    function sweep(address _token) external onlyOwner onlyBeforeDeadline {
         // Do not allow to sweep GRO tokens
-        if (_token == address(GRO)) {
-            revert RedemptionErrors.NoSweepGro();
-        }
+        if (_token == address(GRO)) revert RedemptionErrors.NoSweepGro();
+
         // Transfers the tokens to the owner
-        IERC20(_token).safeTransfer(owner(), IERC20(_token).balanceOf(address(this)));
+        IERC20(_token).safeTransfer(
+            owner(),
+            IERC20(_token).balanceOf(address(this))
+        );
+    }
+
+    // @notice Transfers a portion or all of a user's redeemable GRO position to a new address.
+    // @param to The address to which the GRO position will be transferred.
+    // @param amount The amount of GRO to transfer.
+    function transferPosition(address _to, uint256 _amount) public {
+        if (_amount > (_userGROBalance[msg.sender] - _userClaims[msg.sender]))
```

Comment:
I think this one is mixing apples with oranges? `_userGROBalance` is denominated in GRO tokens, `_userClaims` is denominated in USDC?